apps/legacy_app/notebooks/Analyze_user_behavior_across_different_lending_protocols.ipynb

```diff
@@ -317,33 +355,91 @@
+    "    )\n",
+    "    underlying_addresses_to_decimals = {\n",
+    "        collateral_token_underlying_address: collateral_token_decimals\n",
+    "    }\n",
+    "\n",
+    "    # Fetch prices\n",
+    "    prices = src.helpers.get_prices(token_decimals=underlying_addresses_to_decimals)\n",
+    "    collateral_token_price = prices[collateral_token_underlying_address]\n",
+    "\n",
+    "    return collateral_token_price"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "id": "938f1b8f-eb9c-4508-9fa0-35eca038b3d0",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "{'ETH': 3054.1,\n",
+       " 'WBTC': 76490.05153377741,\n",
+       " 'STRK': 0.42829510632806395,\n",
+       " 'USDC': 1.0050093418439294,\n",
+       " 'USDT': 1.00544949008542}"
+      ]
+     },
+     "execution_count": 6,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "COLLATERAL_TOKENS = [\"ETH\", \"WBTC\", \"STRK\", \"USDC\", \"USDT\"]\n",
+    "prices = {}\n",
+    "for token in COLLATERAL_TOKENS:\n",
+    "    price = fetch_prices(token)\n",
+    "    prices[token] = price\n",
+    "prices"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
    "id": "cc699d21-fde6-4265-93cd-be0970fe97ef",
    "metadata": {},
    "outputs": [
     {
      "name": "stdout",
      "output_type": "stream",
      "text": [
-      "Token Prices in USD: {'dai': {'usd': 1.0}, 'ethereum': {'usd': 2487.73}, 'lords': {'usd': 0.04989956}, 'strike': {'usd': 6.52}, 'tether': {'usd': 0.998271}, 'uno-re': {'usd': 0.01199116}, 'usd-coin': {'usd': 1.0}, 'wrapped-bitcoin': {'usd': 67042}, 'wrapped-steth': {'usd': 2934.02}, 'zenad': {'usd': 0.00118212}}\n"
+      "Token Prices in USD: {'ETH': 3054.1, 'WBTC': 76490.05153377741, 'STRK': 0.42829510632806395, 'USDC': 1.0050093418439294, 'USDT': 1.00544949008542, 'DAI': 0.999747, 'LORDS': 0.04965502, 'WRAPPED-STETH': 3603.56}\n"
      ]
     }
    ],
    "source": [
-    "import requests\n",
-    "\n",
     "# List of token IDs to fetch from CoinGecko (you can add more tokens if needed)\n",
-    "token_ids = 'ethereum,wrapped-bitcoin,usd-coin,dai,tether,wrapped-steth,lords,strike,uno-re,zenad'\n",
+    "token_ids = 'DAI,LORDS,wrapped-steth'\n",
     "\n",
     "# API endpoint\n",
     "url = 'https://api.coingecko.com/api/v3/simple/price'\n",
     "params = {\n",
     "    'ids': token_ids,\n",
     "    'vs_currencies': 'usd'\n",
     "}\n",
-    "\n",
     "response = requests.get(url, params=params)\n",
-    "prices = response.json()\n",
-    "\n",
+    "rem_prices = response.json()\n",
+    "token_ids = token_ids.split(',')\n",
+    "for token, usd in rem_prices.items():\n",
+    "    price = usd['usd']\n",
+    "    for token_id in token_ids:\n",
+    "        if token.lower() == token_id.lower():\n",
+    "            prices[token_id.upper()] = price\n",
```

Comment:
This works good as weel, so pls choose one solution for fetching the prices and remove the other one 🙂